src/chainstate/stacks/boot/costs-2.clar

```diff
@@ -1 +1,567 @@
+    (runtime (linear n u75 u244)))
+
+(define-read-only (cost_as_max_len (n uint))
+    (runtime u475))
+
+(define-read-only (cost_contract_call (n uint))
+    (runtime u153))
+
+(define-read-only (cost_contract_of (n uint))
+    (runtime u13400))
+
+(define-read-only (cost_principal_of (n uint))
+    (runtime u39))
+
+
+(define-read-only (cost_at_block (n uint))
+    {
+        runtime: u210,
+        write_length: u0,
+        write_count: u0,
+        read_count: u1,
+        read_length: u1
+    })
+
+
+(define-read-only (cost_load_contract (n uint))
+    {
+        runtime: (linear n u1 u157),
+        write_length: u0,
+        write_count: u0,
+        ;; set to 3 because of the associated metadata loads
+        read_count: u3,
+        read_length: (linear n u1 u1)
+    })
+
+
+(define-read-only (cost_create_map (n uint))
+    {
+        runtime: (linear n u1 u1631),
+        write_length: (linear n u1 u1),
+        write_count: u1,
+        read_count: u0,
+        read_length: u0
+    })
+
+
+(define-read-only (cost_create_var (n uint))
+    {
+        runtime: (linear n u7 u2152),
+        write_length: (linear n u1 u1),
+        write_count: u2,
+        read_count: u0,
+        read_length: u0
+    })
+
+
+(define-read-only (cost_create_nft (n uint))
+    {
+        runtime: (linear n u1 u1610),
+        write_length: (linear n u1 u1),
+        write_count: u1,
+        read_count: u0,
+        read_length: u0
+    })
+
+
+(define-read-only (cost_create_ft (n uint))
+    {
+        runtime: u1972,
+        write_length: u1,
+        write_count: u2,
+        read_count: u0,
+        read_length: u0
+    })
+
+
+(define-read-only (cost_fetch_entry (n uint))
+    {
+        runtime: (linear n u1 u1539),
+        write_length: u0,
+        write_count: u0,
+        read_count: u1,
+        read_length: (linear n u1 u1)
+    })
+
+
+(define-read-only (cost_set_entry (n uint))
+    {
+        runtime: (linear n u4 u2204),
+        write_length: (linear n u1 u1),
+        write_count: u1,
+        read_count: u1,
+        read_length: u0
+    })
+
+
+(define-read-only (cost_fetch_var (n uint))
+    {
+        runtime: (linear n u1 u543),
+        write_length: u0,
+        write_count: u0,
+        read_count: u1,
+        read_length: (linear n u1 u1)
+    })
+
+
+(define-read-only (cost_set_var (n uint))
+    {
+        runtime: (linear n u5 u691),
+        write_length: (linear n u1 u1),
+        write_count: u1,
+        read_count: u1,
+        read_length: u0
+    })
+
+
+(define-read-only (cost_contract_storage (n uint))
+    {
+        runtime: (linear n u13 u7982),
+        write_length: (linear n u1 u1),
+        write_count: u1,
+        read_count: u0,
+        read_length: u0
+    })
+
+
+(define-read-only (cost_block_info (n uint))
+    {
+        runtime: u6321,
+        write_length: u0,
+        write_count: u0,
+        read_count: u1,
+        read_length: u1
+    })
+
+
+(define-read-only (cost_stx_balance (n uint))
```

Comment:
I'm surprised that getting the STX balance and transferring STX is much more expensive (2x) than doing so for a fungible token.  Do we know why this is?

Reply:
There is more work being done for STX balances -- fetching the current burnchain height and applying any necessary checks for Stacking unlocks to get the current account balance. The difference in runtime of ~700 isn't all that much: that's equivalent to ~4 comparison operations. So this isn't terribly surprising to me given that fetching STX balances requires the additional work on the unlock checks.